code/game/objects/items/melee/baton.dm

```diff
@@ -549,6 +550,16 @@
 	. = ..()
 	if (!(. & EMP_PROTECT_SELF))
 		deductcharge(1000 / severity)
+	if (cell.charge >= cell_hit_cost)
+		for(var/loops = 1, loops < rand(6,12), loops++)
```

Comment:
The rand proc runs on every loop iirc, you should set it to a variable (553-554) so it’s only randomizing once

Reply:
i specifically want it to be randomized every loop